WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsUnknownViewModel.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import Yosemite
+
+final class CardReaderSettingsUnknownViewModel: CardReaderSettingsPresentedViewModel {
+    var shouldShow: Bool
+    var didChangeShouldShow: ((Bool) -> Void)?
+
+    private var noConnectedReaders: Bool
+    private var noKnownReaders: Bool
+
+    init(didChangeShouldShow: ((Bool) -> Void)?) {
+
+        // Default shouldShow to true, since we don't know if there are any known or connected readers
+        // TODO: Decide if we want a flag to reflect this initial state and use it to drive a loading indicator
+
+        self.shouldShow = true
+        self.didChangeShouldShow = didChangeShouldShow
+        self.noConnectedReaders = true
+        self.noKnownReaders = true
+
+        beginObservation()
+    }
+
+    /// Dispatches actions to the CardPresentPaymentStore so that we can monitor changes to the list of known
+    /// and connected readers.
+    ///
+    private func beginObservation() {
+
+        // This completion should be called repeatedly as the list of known readers changes
+        let knownAction = CardPresentPaymentAction.observeKnownReaders() { [weak self] readers in
+            self?.noKnownReaders = readers.isEmpty
+            self?.reevaluateShouldShow()
+        }
+        ServiceLocator.stores.dispatch(knownAction)
+
+        // This completion should be called repeatedly as the list of connected readers changes
+        let connectedAction = CardPresentPaymentAction.observeConnectedReaders() { [weak self] readers in
+            self?.noConnectedReaders = readers.isEmpty
```

Comment:
This is a personal preference, so please take it as that (meaning, feel free to ignore this)

I find the self? a bit annoying, specially where there are several lines attacking self? so I tend to do something like 

```
guard let self = self else { return }
self.oneThing()
self.anotherThing()
```

I could argue that's also a bit of a "performance" improvement, as the execution would short-circuit if there is no valid self at that point, but meh, that'd be a pretty weak argument anyway. 🤷🏼

Reply:
Updated in https://github.com/woocommerce/woocommerce-ios/pull/4081/commits/06a4be7721bde38a2117f137a6b4b08874bb237f